spark/src/main/java/org/apache/iceberg/actions/BaseSparkAction.java

```diff
@@ -137,7 +138,19 @@
     }
     // Try catalog based name based resolution
     try {
-      return spark.table(tableName + "." + type);
+      if (tableName.startsWith("spark_catalog")) {
+        // Do to the design of Spark, we cannot pass multi element namespaces to the session catalog
+        // We also don't know whether the Catalog is Hive or Hadoop Based, so we will try to load it
+        // in the hive manner first, then fall back and try the location if we have completely run out of options
+        // TODO remove this when we have Spark workaround for multipart identifiers in SparkSessionCatalog
+        try {
+          return noCatalogReader.load(tableName.replaceFirst("spark_catalog\\.", "") + "." + type);
```

Comment:
Would it make sense to call it `dataFrameReader` instead of `noCatalogReader` everywhere?

Reply:
:shrug: It's a no catalog reader whenever it's used :) But sure I can rename it.